mmcv/transforms/formatting.py

```diff
@@ -0,0 +1,603 @@
+            non_prob = 1 - self.prob
+            # exclude non-flip
+            single_ratio = self.prob / (len(direction_list) - 1)
+            prob_list = [single_ratio] * (len(direction_list) - 1) + [non_prob]
+
+        cur_dir = np.random.choice(direction_list, p=prob_list)
+
+        return cur_dir
+
+    def _flip(self, results: dict) -> None:
+        """Resize images, bounding boxes, semantic segmentation map and
+        keypoints."""
+        # flip image
+        results['img'] = mmcv.imflip(
+            results['img'], direction=results['flip_direction'])
+
+        img_shape = results['img'].shape[:2]
+
+        # flip bboxes
+        if results.get('gt_bboxes', None) is not None:
+            results['gt_bboxes'] = self.bbox_flip(results['gt_bboxes'],
+                                                  img_shape,
+                                                  results['flip_direction'])
+
+        # flip keypoints
+        if results.get('gt_keypoints', None) is not None:
+            results['gt_keypoints'] = self.keypoints_flip(
+                results['gt_keypoints'], img_shape, results['flip_direction'])
+
+        # flip segs
+        if results.get('gt_semantic_seg', None) is not None:
+            results['gt_semantic_seg'] = mmcv.imflip(
+                results['gt_semantic_seg'],
+                direction=results['flip_direction'])
+
+    def _flip_with_override(self, results: dict) -> None:
+        """Function to flip flip images, bounding boxes, semantic segmentation
+        map and keypoints, when `override` is set to `True`"""
+        cur_dir = self._choose_direction()
+        if cur_dir is None:
+            results['flip'] = False
+            results['flip_direction'] = None
+        else:
+            results['flip'] = True
+            results['flip_direction'] = cur_dir
+            self._flip(results)
+
+    def transform(self, results: dict) -> dict:
+        """Transform function to flip images, bounding boxes, semantic
+        segmentation map and keypoints.
+
+        Args:
+            results (dict): Result dict from loading pipeline.
+        Returns:
+            dict: Resized results, 'img', 'gt_bboxes', 'gt_semantic_seg',
+                'gt_keypoints', 'flip', and 'flip_direction' keys are
+                updated in result dict.
+        """
+        if self.override:
+            self._flip_with_override(results)
+        else:
+            if 'flip' in results and results['flip']:
+                if 'flip_direction' in results and results[
+                        'flip_direction'] is not None:
+                    self._flip(results)
+                else:
+                    self._flip_with_override(results)
+            else:
+                self._flip_with_override(results)
+
+        return results
+
+    def __repr__(self) -> None:
+        return self.__class__.__name__ + f'(prob={self.prob})'
+
+
+@TRANSFORMS.register_module()
+class RandomResize(BaseTransform):
```

Comment:
RandomResize should not be in formatting.py. Considering put these transforms into geometric.py.